microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIConfigFileHealthTest.java

```diff
@@ -0,0 +1,204 @@
+ * {@link FailSafeDummyService} configured at the beginning via MP Config properties:
+ * <ul>
+ * <li>{@link FailSafeDummyService#LIVE_CONFIG_PROPERTY} - service is live</li>
+ * <li>{@link FailSafeDummyService#READY_CONFIG_PROPERTY} - service is ready if
+ * {@link FailSafeDummyService#IN_MAINTENANCE_CONFIG_PROPERTY} is {@code false}</li>
+ * <li>{@link FailSafeDummyService#READY_IN_MAINTENANCE_CONFIG_PROPERTY} - service is ready if
+ * {@link FailSafeDummyService#IN_MAINTENANCE_CONFIG_PROPERTY} is {@code true}</li>
+ * </ul>
+ * If service is in maintenance a method simulating opening resources {@link FailSafeDummyService#simulateOpeningResources}
+ * throws and {@link IOException}.
+ *
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(MicroProfileFTSetupTask.class)
+public class FailSafeCDIConfigFileHealthTest {
+
+    private static final String MPConfigContent = String.format("%s=%s\n%s=%s\n%s=%s\n%s=%s",
+            FailSafeDummyService.LIVE_CONFIG_PROPERTY, false,
+            FailSafeDummyService.READY_CONFIG_PROPERTY, true,
+            FailSafeDummyService.IN_MAINTENANCE_CONFIG_PROPERTY, true,
+            FailSafeDummyService.READY_IN_MAINTENANCE_CONFIG_PROPERTY, false);
+
+    protected RequestSpecification metricsRequest;
+
+    @ArquillianResource
+    protected ManagementClient managementClient;
+
+    @Before
+    public void before() throws ConfigurationException, InterruptedException, TimeoutException, IOException {
+        ArquillianContainerProperties arqProps = new ArquillianContainerProperties(
+                ArquillianDescriptorWrapper.getArquillianDescriptor());
+        String url = "http://" + arqProps.getDefaultManagementAddress() + ":" + arqProps.getDefaultManagementPort()
+                + "/metrics";
+        metricsRequest = given().baseUri(url).accept(ContentType.JSON);
+        // reset MP Config properties
+        new Administration(ManagementClientProvider.onlineStandalone(managementClient)).reload();
+    }
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        return ShrinkWrap
+                .create(WebArchive.class, FailSafeCDIHealthBaseTest.class.getSimpleName() + ".war")
+                .addClasses(FailSafeDummyService.class, CDIBasedLivenessHealthCheck.class, CDIBasedReadinessHealthCheck.class)
+                .addAsManifestResource(new StringAsset(MPConfigContent), "microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    /**
+     * @tpTestDetails Multiple-component customer scenario - health check status is based on fail-safe CDI bean and MP Config
+     *                property. There is IOException since service is in maintenance but fallback method should be used.
+     * @tpPassCrit Overall and the health check status is down and fallback method is invoked. MP Metrics are increased
```

Comment:
Rewording needed? `Overall and the health check status is down` -> `Overall and health check status are down`.
The same applies to the following test methods Javadoc, too.
WDYT?